flink-python/pyflink/table/query_config.py

```diff
@@ -0,0 +1,112 @@
+        super(StreamQueryConfig, self).__init__(self._j_stream_query_config)
+
+    def with_idle_state_retention_time(self, min_time, max_time):
+        """
+        Specifies a minimum and a maximum time interval for how long idle state, i.e., state which
+        was not updated, will be retained.
+
+        State will never be cleared until it was idle for less than the minimum time and will never
+        be kept if it was idle for more than the maximum time.
+
+        When new data arrives for previously cleaned-up state, the new data will be handled as if it
+        was the first data. This can result in previous results being overwritten.
+
+        Set to ``datetime.timedelta()``(zero) to never clean-up the state.
+
+        .. note::
+            Cleaning up state requires additional bookkeeping which becomes less expensive for
+            larger differences of minTime and maxTime. The difference between minTime and maxTime
+            must be at least ``datetime.timedelta(minutes=5)``(5 minutes).
+
+        :param min_time: The minimum time interval for which idle state is retained. Set to
+                         ``datetime.timedelta()``(zero) to never clean-up the state.
+        :param max_time: The maximum time interval for which idle state is retained. Must be at
+                         least 5 minutes greater than minTime. Set to
+                         ``datetime.timedelta()``(zero) to never clean-up the state.
+        :return: :class:`StreamQueryConfig`
+        """
+        #  type: (timedelta, timedelta) -> StreamQueryConfig
+        j_time_class = self._jvm.org.apache.flink.api.common.time.Time
+        j_min_time = j_time_class.milliseconds(long(round(min_time.total_seconds() * 1000)))
+        j_max_time = j_time_class.milliseconds(long(round(max_time.total_seconds() * 1000)))
+        self._j_stream_query_config = \
+            self._j_stream_query_config.withIdleStateRetentionTime(j_min_time, j_max_time)
+        return self
+
+    def get_min_idle_state_retention_time(self):
+        """
+        State might be cleared and removed if it was not updated for the defined period of time.
+
+        :return: The minimum time until state which was not updated will be retained.
+        """
+        #  type: () -> int
+        return self._j_stream_query_config.getMinIdleStateRetentionTime()
+
+    def get_max_idle_state_retention_time(self):
+        """
+        State will be cleared and removed if it was not updated for the defined period of time.
+
+        :return: The maximum time until state which was not updated will be retained.
+        """
+        #  type: () -> int
+        return self._j_stream_query_config.getMaxIdleStateRetentionTime()
+
+
+class BatchQueryConfig(QueryConfig):
+    """
+    The :class:`BatchQueryConfig` holds parameters to configure the behavior of batch queries.
+    """
+
+    def __init__(self, j_batch_query_config=None):
+        self._jvm = get_gateway().jvm
```

Comment:
There is no need to make `_jvm` an instance variable

Reply:
Done.